app/assets/javascripts/authorize_account.js

```diff
@@ -0,0 +1,43 @@
+class AuthorizeAccount {
+  constructor(emailField, pubKey, domain) {
+    this.emailField = emailField;
+    this.pubKey = pubKey;
+    this.domain = domain;
+  }
+
+  async accessRequest() {
+    console.log('Modified')
+    const boltEmbedded = Bolt(this.pubKey);
+
+    if (this.emailField) {
+      const email = this.emailField.value
+      const encodedEmail = encodeURIComponent(email);
+      // Call the endpoint to check if the email already exist on bolt
+      const response = await fetch(this.domain + "/v1/account/exists?email=" + encodedEmail);
+      const responseAsJson = await response.json();
+
+      if(responseAsJson.has_bolt_account) {
+        this.emailField.parentNode.setAttribute("class", "email-div")
+        let authorizationComponent = boltEmbedded.create("authorization_component", {style: "callout"});
+        await authorizationComponent.mount(".email-div");
+        // start OTP modal
+        let authorizationResponse = await authorizationComponent.authorize({"email": email});
+        if (authorizationResponse) {
+
+          const authorizationCode = authorizationResponse.authorizationCode;
+          const scope = authorizationResponse.scope ;
+
+          document.location.href = `/users/auth/bolt?authorization_code=${authorizationCode}&scope=${scope}`
+        }
+      }
+    }
+  }
+
+  async hasBoltAccount(email) {
+    const encodedEmail = encodeURIComponent(email);
+    // Call the endpoint to check if the email already exist on bolt
+    const response = await fetch(this.domain + "/v1/account/exists?email=" + encodedEmail);
+    const responseAsJson = await response.json();
+    return responseAsJson.has_bolt_account
+  }
```

Comment:
is this used somewhere?

Reply:
The `hasBoltAccount` method checks if the email entered exists on Bolt or not. Depending on the response we show the login authentication modal

Reply:
Oh I see
Removed this now

Reply:
check here [dc7e3f5](https://github.com/nebulab/bolt_demo/commit/dc7e3f594070c4d9533b805f52ee0da9283907af)